mmdet3d/models/backbones/dla.py

```diff
@@ -0,0 +1,313 @@
+        34: (BasicBlock, (1, 1, 1, 2, 2, 1), (16, 32, 64, 128, 256, 512)),
+    }
+
+    def __init__(self,
+                 depth=34,
+                 in_channels=3,
+                 norm_cfg=None,
+                 residual_root=False):
+        super(DLANet, self).__init__()
+        if depth not in self.arch_settings:
+            raise KeyError(f'invalida depth {depth} for DLA')
+        block, levels, channels = self.arch_settings[depth]
+        self.channels = channels
+        # self.num_classes = num_classes
+        self.base_layer = nn.Sequential(
+            nn.Conv2d(
+                in_channels,
+                channels[0],
+                kernel_size=7,
+                stride=1,
+                padding=3,
+                bias=False),
+            dla_build_norm_layer(norm_cfg, channels[0])[1],
+            nn.ReLU(inplace=True))
+        self.level0 = self._make_conv_level(channels[0], channels[0],
+                                            levels[0], norm_cfg)
+        self.level1 = self._make_conv_level(
+            channels[0], channels[1], levels[1], norm_cfg, stride=2)
+        self.level2 = Tree(
+            levels[2],
+            block,
+            channels[1],
+            channels[2],
+            norm_cfg,
+            2,
+            level_root=False,
+            root_residual=residual_root)
+        self.level3 = Tree(
+            levels[3],
+            block,
+            channels[2],
+            channels[3],
+            norm_cfg,
+            2,
+            level_root=True,
+            root_residual=residual_root)
+        self.level4 = Tree(
+            levels[4],
+            block,
+            channels[3],
+            channels[4],
+            norm_cfg,
+            2,
+            level_root=True,
+            root_residual=residual_root)
+        self.level5 = Tree(
+            levels[5],
+            block,
+            channels[4],
+            channels[5],
+            norm_cfg,
+            2,
+            level_root=True,
+            root_residual=residual_root)
+
+    def _make_conv_level(self,
+                         inplanes,
+                         planes,
+                         convs,
+                         norm_cfg,
+                         stride=1,
+                         dilation=1):
+        modules = []
+        for i in range(convs):
+            modules.extend([
+                nn.Conv2d(
+                    inplanes,
+                    planes,
+                    kernel_size=3,
+                    stride=stride if i == 0 else 1,
+                    padding=dilation,
+                    bias=False,
+                    dilation=dilation),
+                dla_build_norm_layer(norm_cfg, planes)[1],
+                nn.ReLU(inplace=True)
+            ])
+            inplanes = planes
+        return nn.Sequential(*modules)
+
+    def forward(self, x):
+        y = []
+        x = self.base_layer(x)
+        for i in range(6):
```

Comment:
`6` is a bot hard-code. Can we infer the number from e.g. `len(self.levels)`? We may set `self.layers = [self.layer1, self.layer2, ..., self.layer6]` in `__init__`.

Reply:
I made some refinements like you said, but a little different.

Reply:
Yes that's acceptable :)